cloud_experts_tutorials/cloud-experts-getting-started/cloud-experts-getting-started-openshift-concepts.adoc

```diff
@@ -0,0 +1,88 @@
+
+For example, to create a reproducible build pipeline for Tomcat and Maven:
+
+. Create a builder image containing OpenJDK and Tomcat that expects to have a WAR file injected.
+
+. Create a second image that layers on top of the first image Maven and any other standard dependencies, and expects to have a Maven project injected.
+
+. Start S2I using the Java application source and the Maven image to create the desired application WAR.
+
+. Start S2I a second time using the WAR file from the earlier step and the initial Tomcat image to create the runtime image.
+
+By placing build logic inside of images and combining the images into multiple steps, the runtime environment is close to the build environment without requiring the deployment of build tools to production.
+
+=== S2I benefits
+
+==== Reproducibility
+Allow build environments to be tightly versioned by encapsulating them within a container image and defining a simple interface of injected source code for callers. Reproducible builds are a key requirement for enabling security updates and continuous integration in containerized infrastructure, and builder images help ensure repeatability and the ability to swap run times.
+
+==== Flexibility
+Any existing build system that can run on Linux can run inside of a container, and each individual builder can also be part of a larger pipeline. The scripts that process the application source code can be injected into the builder image, allowing authors to adapt existing images to enable source handling.
+
+==== Speed
+Instead of building multiple layers in a single Dockerfile, S2I encourages authors to represent an application in a single image layer. This saves time during creation and deployment and allows for better control over the output of the final image.
+
+==== Security
+Dockerfiles are run without many of the normal operational controls of containers. They usually run as root and have access to the container network. S2I can control what permissions and privileges are available to the builder image since the build is launched in a single container. In concert with platforms like OpenShift, S2I allows administrators to control what privileges developers have at build time.
+
+== Routes
+An OpenShift Route exposes a service at a hostname so that external clients can reach it by name. When a Route object is created on OpenShift, it gets picked up by the built-in HAProxy load balancer to expose the requested service and make it externally available with the given configuration. Similar to the Kubernetes Ingress object, Red Hat created the concept of Route to fill a need and then contributed the design principles behind it to the community, which heavily influenced the Ingress design. A Route does have some additional features as can be seen in the following chart.
```

Comment:
* Consider [repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting) on API object references (general vs object) re: "route" and "ingress"
* Maybe break up the para
* End with colon
```suggestion
An OpenShift route exposes a service at a hostname so that external clients can reach it by name. When a `Route` object is created on OpenShift, it gets picked up by the built-in HAProxy load balancer to expose the requested service and make it externally available with the given configuration.

Similar to the Kubernetes `Ingress` object, Red Hat created the concept of a route to fill a need and then contributed the design principles behind it to the community, which heavily influenced the `Ingress` object design. A route does have some additional features, as can be seen in the following chart:
```